File: src/test/java/com/github/xumeng/java/problems/easy/problem0344/Solution0344Test.java
package com.github.xumeng.java.problems.easy.problem0344;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0344Test {

    private final Solution0344 solution = new Solution0344();

    @Test
    public void test1() {
        char[] s = {'a'};
        solution.reverseString(s);
        char[] expected = {'a'};
        assertArrayEquals(expected, s);
    }

    @Test
    public void test2() {
        char[] s = {'h', 'e', 'l', 'l', 'o'};
        solution.reverseString(s);
        char[] expected = {'o', 'l', 'l', 'e', 'h'};
        assertArrayEquals(expected, s);
    }

    @Test
    public void test3() {
        char[] s = {'H', 'a', 'n', 'n', 'a', 'h'};
        solution.reverseString(s);
        char[] expected = {'h', 'a', 'n', 'n', 'a', 'H'};
        assertArrayEquals(expected, s);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0344/Solution0344Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0344

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0344Test {

    private val solution = Solution0344()

    @Test
    fun test1() {
        val s = charArrayOf('a')
        solution.reverseString(s)
        val expected = charArrayOf('a')
        assertArrayEquals(expected, s)
    }

    @Test
    fun test2() {
        val s = charArrayOf('h', 'e', 'l', 'l', 'o')
        solution.reverseString(s)
        val expected = charArrayOf('o', 'l', 'l', 'e', 'h')
        assertArrayEquals(expected, s)
    }

    @Test
    fun test3() {
        val s = charArrayOf('H', 'a', 'n', 'n', 'a', 'h')
        solution.reverseString(s)
        val expected = charArrayOf('h', 'a', 'n', 'n', 'a', 'H')
        assertArrayEquals(expected, s)
    }

}